src/persistence.rs

```diff
@@ -246,26 +235,26 @@ pub mod embedding {
     impl EmbeddingPersistor for TextFileVectorPersistor {
         fn put_metadata(&mut self, entity_count: u32, dimension: u16) {
             let metadata = format!("{} {}", entity_count, dimension);
-            self.buf_writer.write(metadata.as_bytes());
+            self.buf_writer.write(metadata.as_bytes()).ok();
         }
 
         fn put_data(&mut self, entity: String, occur_count: u32, vector: Vec<f32>) {
-            self.buf_writer.write(b"\n");
-            self.buf_writer.write(entity.as_bytes());
+            self.buf_writer.write(b"\n").ok();
```

Comment:
We can return errors. Maybe, change return types to `Result` so then we can just collect all entities with problems and log (to console or file).
`ok` doesn't make sense because only make `clippy` happy :)